zenodo/modules/records/models.py

```diff
@@ -68,6 +88,9 @@ class AccessRight(object):
     @staticmethod
     def is_embargoed(embargo_date):
         """Test if date is still under embargo."""
+        embargo_date = embargo_date \
+            if not isinstance(embargo_date, string_types) \
+            else parser.parse(embargo_date).date()
```

Comment:
For date parsing it's better to use arrow library for consistent among our modules (under the hood it uses dateutil):
```python
import arrow
arrow.get(value).date()
```